contributing.md

```diff
@@ -0,0 +1,163 @@
+* `mediafile`
+* `jellyfish`
+
+Please be aware that this list might be out of date, but [setup.py](https://github.com/beetbox/beets/blob/60bba370c0ecc58b4057d78867273e5936d22dee/setup.py#L89-L110) contains a list that is always up to date. With all the required modules in place, you can just type `./beet` in the source directory to run beets from there.
+
+
+### Code contribution ideas
+* We maintain a set of [issues marked as "bite-sized"](https://github.com/beetbox/beets/labels/bitesize). These are issues that would serve as a good introduction to the codebase. Claim one and start exploring!
+* Our [test coverage](https://codecov.io/github/beetbox/beets) is somewhat low. You can help out by finding low-coverage modules on Coveralls and adding new tests.
+* There are several ways to improve the tests in general (see [Testing](https://github.com/beetbox/beets/wiki/Testing)) and some places to think about performance optimization (see [Optimization](https://github.com/beetbox/beets/wiki/Optimization)).
+* Not all of our code is up to our coding conventions. In particular, the [API documentation][apidocs] are currently quite sparse. You can help by adding to the docstrings in the code and to the documentation pages themselves.
+
+## Your first contribution
+If this is your first time contributing to an open source project, welcome! If you are confused at all about how to contribute or what to contribute, take a look at [this great tutorial](http://makeapullrequest.com/), or stop by our [gitter chat room](https://gitter.im/beetbox/beets#) if you have any questions. 
+
+We maintain a list of issues we feel are perfect for first-timers tagged "[good first issue](https://github.com/beetbox/beets/labels/good%20first%20issue)". These issues are perfect for anyone new to open source. In fact, we encourage only users who are new to open source to tackle these. Ideally, the goal for these issues is to get users comfortable with contributing to an open source project for the first time. Please, do not hesitate to ask any questions.
+
+## How to Submit Your Work
+
+Do you have a great bug fix, new feature, or documentation expansion you'd like to contribute? Follow these steps to create a GitHub pull request and your code will ship in no time.
+
+1. Fork the beets repository and clone it (see above) to create a workspace.
+2. Make your changes.
+3. Add tests. If you've fixed a bug, write a test to ensure that you've actually fixed it. If there's a new feature or plugin, please contribute tests that show that your code does what it says.
+4. Add documentation. If you've added a new command flag, for example, find the appropriate page under `docs/` where it needs to be listed.
+5. Add a changelog entry to `docs/changelog.rst` near the top of the document.
+6. Run the tests and style checker. The easiest way to run the full battery of tests is to use [tox](https://tox.readthedocs.org/en/latest/). This will run the same set of tests that we run on Travis.
```

Comment:
I noticed errors running tox and had to figure out how to install the specific python versions it requires for it's tests. Maybe the best place to explain that nuance is in the Testing wiki and then re-linking that page here?

Reply:
Yes, good point. We also have a [wiki page on testing](https://github.com/beetbox/beets/wiki/Testing) that should perhaps have some of its information collapsed here.

I don't know of a succinct description to link to, but here's a shot at capturing what we need to say:

> Tox tries to test on multiple Python versions by default. You'll see errors for the Python versions you don't have installed. It can be expedient to run the tests only on the version(s) of Python you already have. Use `tox -e py38-test` to only run the tests on Python 3.8, for example.

Reply:
I'll try to clarify more in the wiki page and just link to it here